### PR TITLE
fix(mock): Swallow stack trace on an un-initialized project upon `amplify mock api`

### PR DESCRIPTION
**Pull request by manaswi223**
#### Issue #, if available
https://github.com/aws-amplify/amplify-category-api/issues/1324

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli-core/src/errors/amplify-error.ts [hunk 1]
```diff
@@ -20,5 +20,6 @@ export class AmplifyError extends AmplifyException {
    */
   constructor(name: AmplifyErrorType, options: AmplifyExceptionOptions, downstreamException?: Error) {
     super(name, 'ERROR', options, downstreamException);
+    this.stack = undefined;
```
Comment (sobolk) [inline packages/amplify-cli-core/src/errors/amplify-error.ts:23]: we should not erase stack. we use that in telemetry

File: packages/amplify-cli-core/src/errors/amplify-exception.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export abstract class AmplifyException extends Error {
   public readonly details?: string;
   public readonly link?: string;
   public readonly code?: string;
+  public readonly cause?: Error;
```
Comment (sobolk) [inline packages/amplify-cli-core/src/errors/amplify-exception.ts:13]: this property is redundant with `downstreamException`. Please remove it.

File: packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.ts [hunk 3]
```diff
@@ -3,12 +3,23 @@ import { $TSAny, AmplifyError, stateManager } from 'amplify-cli-core';
...
-  });
+    cause: new Error('`amplify init` not done'),
+  }); 
+  // handle error message gracefully by swallowing the stack trace
+} catch(error) {
+  if (!exceptionPrinted) {
+    exceptionPrinted = true;
+    console.error(`Error: ${error.message}\nResolution: ${error.resolution}`);
+  } 
+}
```
Comment (sobolk) [inline packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.ts:24]: We should not make this change here. `getEnvInfo` is called in multiple scenarios - we should not swallow the exception there.
If you're attempting to fix amplify mock api then I suggest you catch that error in the mock package and apply special handling there.

**Review by sobolk — CHANGES_REQUESTED** [review]

File: packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.ts (hunk 3 above)
Comment (manaswi223) [reply packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.ts:24]: yes, I am attempting to make changes to the mock. Sure, thanks @sobolk, will make changes there : )

**Pull request closed without merging**